network/default.go

```diff
@@ -39,11 +50,18 @@ type network struct {
 	// client is network client
 	client client.Client
 
+	// ctrlClient is ControlChannel client
+	ctrlClient transport.Client
+	// netClient is NetwrokChannel client
+	netClient transport.Client
+
 	sync.RWMutex
 	// connected marks the network as connected
 	connected bool
 	// closed closes the network
 	closed chan bool
+	// neighbours maps the node neighbourhood
+	neighbours map[node]map[node]bool
```

Comment:
Hmm. Whenever I see map in a map it makes me uncomfortable. Highlights the need for a struct. Maybe store as `map[string]*node` with the below struct and node id as key.

```
type node struct {
  id string
  address string
  neighbours map[string]*node
}
```

Reply:
This is normal thing to do in other languages (Python, Ruby and Rust), including [Go](https://blog.golang.org/go-maps-in-action).

Using `node` as a key makes it easy to iterate over and perform ops on the key object, but I'm happy to rework this of course.